bukkit/src/main/java/com/github/juliarn/npclib/bukkit/BukkitActionController.java

```diff
@@ -170,6 +197,32 @@ public void handleMove(@NotNull PlayerMoveEvent event) {
     }
   }
 
+  @EventHandler
+  public void onChunkLoad(ChunkLoadEvent event) {
+    World world = event.getWorld();
+    Chunk chunk = event.getChunk();
+
+    // Add the chunk to the cache
+    loadedChunks
+      .computeIfAbsent(world, w -> new HashSet<>())
+      .add(chunkKey(chunk.getX(), chunk.getZ()));
```

Comment:
I would advise just using `Chunk.getChunkKey()` and storing the long this provides. It is much more efficient than constructing a string on every single chunk load and unload

Reply:
I don't think `getChunkKey` is an api that is available since 1.8, let alone in spigot/bukkit.

Reply:
It's available in the paper-api since I think about 1.12, but all the same, here's the method: 

```
static long getChunkKey(int x, int z) {
    return (long)x & 4294967295L | ((long)z & 4294967295L) << 32;
} 
```